x-pack/plugins/security_solution/public/management/pages/policy/view/tabs/policy_tabs.tsx

```diff
@@ -70,40 +69,57 @@ interface PolicyTab {
 export const PolicyTabs = React.memo(() => {
   const history = useHistory();
   const http = useHttp();
+  const toasts = useToasts();
+
   const isInSettingsTab = usePolicyDetailsSelector(isOnPolicyFormView);
   const isInTrustedAppsTab = usePolicyDetailsSelector(isOnPolicyTrustedAppsView);
-  const isInEventFilters = usePolicyDetailsSelector(isOnPolicyEventFiltersView);
+  const isInEventFiltersTab = usePolicyDetailsSelector(isOnPolicyEventFiltersView);
   const isInHostIsolationExceptionsTab = usePolicyDetailsSelector(isOnHostIsolationExceptionsView);
   const isInBlocklistsTab = usePolicyDetailsSelector(isOnBlocklistsView);
   const policyId = usePolicyDetailsSelector(policyIdFromParams);
   const policyItem = usePolicyDetailsSelector(policyDetails);
-  const privileges = useUserPrivileges().endpointPrivileges;
+  const {
+    canReadTrustedApplications,
+    canWriteTrustedApplications,
+    canReadEventFilters,
+    canWriteEventFilters,
+    canReadHostIsolationExceptions,
+    canWriteHostIsolationExceptions,
+    canReadBlocklist,
+    canWriteBlocklist,
+    loading: privilegesLoading,
+  } = useUserPrivileges().endpointPrivileges;
   const { state: routeState = {} } = useLocation<PolicyDetailsRouteState>();
 
-  const allPolicyHostIsolationExceptionsListRequest = useListArtifact(
-    HostIsolationExceptionsApiClient.getInstance(http),
-    {
-      page: 1,
-      perPage: 100,
-      policies: [policyId, 'all'],
-    },
-    HOST_ISOLATION_EXCEPTIONS_SEARCHABLE_FIELDS,
-    {
-      enabled: !privileges.loading && !privileges.canIsolateHost,
-    }
-  );
-
-  const canSeeHostIsolationExceptions =
-    privileges.canIsolateHost ||
-    (allPolicyHostIsolationExceptionsListRequest.isFetched &&
-      allPolicyHostIsolationExceptionsListRequest.data?.total !== 0);
-
-  // move the use out of this route if they can't access it
+  // move the user out of this route if they can't access it
   useEffect(() => {
-    if (isInHostIsolationExceptionsTab && !canSeeHostIsolationExceptions) {
+    if (
+      (isInTrustedAppsTab && !canReadTrustedApplications) ||
+      (isInEventFiltersTab && !canReadEventFilters) ||
+      (isInHostIsolationExceptionsTab && !canReadHostIsolationExceptions) ||
+      (isInBlocklistsTab && !canReadBlocklist)
+    ) {
       history.replace(getPolicyDetailPath(policyId));
+      toasts.addDanger(
+        i18n.translate('xpack.securitySolution.policyDetails.missingArtifactAccess', {
+          defaultMessage:
+            'You do not have the required Kibana permissions to use artifacts by policy',
```

Comment:
@kevinlog Let me know what do you think about this message

Reply:
For me, this is a bit misleading. It is not about using artifacts _by policy_, it's about seeing a given artifact.

So if you have TrustedApps:READ, but Blocklists:NONE and open an url for blocklists, you see this message, even though you can see TrustedApps. I'd go with something like: _'You do not have the required Kibana permissions to use the given artifact.'_

(Also, with a period on the end.)

@dasansol92 @kevinlog what do you think?

Reply:
yeah, I agree with @gergoabraham  here.